lib/environment.js

```diff
@@ -8,94 +8,35 @@ var ENV = function(opts) {
   this.rootPath = opts.rootPath || '.';
   this.sourcePath = opts.sourcePath || 'source';
   this.ignorePrefixPattern = opts.ignorePrefixPattern || new RegExp('\/_');
-
-  this.trail = new hike.Trail(this.rootPath);
-  this.trail.paths.append(this.sourcePath);
 };
 
-ENV.prototype.getSourceFiles = function(whitelist) {
+ENV.prototype.getSourceFiles = function(whitelist, callback) {
   var _this = this;
   whitelist = whitelist || [];
 
-  return this.getSourcePaths(whitelist).map(function(sourcePath){
-    return new SourceFile({
-      name: path.basename(sourcePath),
-      path: sourcePath
-    });
-  });
-};
-
-ENV.prototype.getSourcePaths = function(whitelist) {
-  var _this = this;
-  var paths = [];
-
-  // If we are requesting only one path, let's speed things up
-  // by using the trail.find method instead of traversing the 
-  // entire tree.
-  if(whitelist.length == 1) {
-    var singlePath = this.trail.find(whitelist[0]);
-    if(!!singlePath) {
-      paths.push(path.relative(this.rootPath, singlePath));
-    }
-
-    return paths;
-  };
-
-  this.eachEntry(this.sourcePath, function(filePath) {
-    if (_this.trail.stat(filePath).isDirectory()) {
-      return;
-    }
-
-    // Should the path be ignored based on the ignorePrefix?
-    var matches = _this.ignorePrefixPattern.exec(filePath);
-    if (matches) {
-      return;
-    }
-
-    // Should the path be filtered out because it's not 
-    // in the requested paths?
-    var whitelisted = true;
+  recursiveReaddir(_this.sourcePath, ['_*'], function (err, files) {
```

Comment:
I fixed this in the latest version. Had to modify the recursive-readdir library (and vendor it) since it did not support RegExp objects. It would be a breaking change in Lingon to switch from RegExp objects to glob patterns for the ignorePrefixPattern value. 

I'll work on a proposed PR for the recursive-readdir project to allow both RegExp instances and globs.